frontend/src/components/node/Details.tsx

```diff
@@ -88,11 +117,23 @@ export default function NodeDetails() {
               <IconButton
                 aria-label={cordonOrUncordon}
                 onClick={() => handleNodeScheduleState(item, cordon)}
+                disabled={isupdatingNodeScheduleProperty}
               >
                 {cordon ? <Icon icon="mdi:check-circle-outline" /> : <Icon icon="mdi:cancel" />}
               </IconButton>
             </Tooltip>
           </AuthVisible>,
+          <AuthVisible item={item} authVerb="delete">
+            <Tooltip title={t('Drain')}>
+              <IconButton
+                aria-label={t('Drain')}
+                onClick={() => handleNodeDrain(item)}
+                disabled={isNodeDrainInProgress}
+              >
+                <Icon icon="mdi:delete-variant" />
+              </IconButton>
```

Comment:
Maybe this can be made simpler with the _ActionButton_.

Reply:
updated